flink-runtime/src/main/java/org/apache/flink/runtime/shuffle/PartitionDescriptor.java

```diff
@@ -53,14 +54,22 @@ public class PartitionDescriptor implements Serializable {
     /** Connection index to identify this partition of intermediate result. */
     private final int connectionIndex;
 
+    /** Whether the intermediate result is a broadcast result. */
+    private final boolean isBroadcast;
+
+    /** The distribution pattern of the intermediate result. */
+    private final DistributionPattern distributionPattern;
```

Comment:
I would suggest to use a boolean type instead of DistributionPattern to avoid exposing a the new internal type which is not public. Currently, this new information is never used in the NettyShuffleMaster, if someone modifies the DistributionPattern, for example, removing some enum fields, the compatibility will be broken and the person who modifies DistributionPattern will never realize that. I think we can use something like this to replace the current field:
```private final boolean isAllToAllDistribution;```
then we only rely on the all to all property and we convert the DistributionPattern to isAllToAllDistribution internally, if someone modifies the code, then he/she must also take care of the internal conversion logic. What do you think?

Reply:
@wsry Good point. Thanks for reviewing the code. I have updated the PR according to the comments, could you please take a look again?